src/App/pages/RecipesList/components/Pagination/Pagination.tsx

```diff
@@ -27,19 +30,63 @@ const Pagination: React.FC<PaginationProps> = ({
   // Если страница всего одна, не показываем пагинацию
   if (totalPages <= 1) return null;
 
+  const pages: (number | string)[] = [];
+
+  // Добавляем первую страницу
+  pages.push(1);
+
+  // Добавляем '...' если текущая страница больше 3
+  if (currentPage > 3) {
+    pages.push('...');
+  }
+
+  // Добавляем страницы вокруг текущей
+  for (let i = Math.max(2, currentPage - 1); i <= Math.min(currentPage + 1, totalPages - 1); i++) {
+    pages.push(i);
+  }
+
+  // Добавляем '...' если текущая страница меньше максимальной
+  if (currentPage < totalPages - 2) {
+    pages.push('...');
+  }
+
+  // Добавляем последнюю страницу
+  if (totalPages > 1) {
+    pages.push(totalPages);
+  }
+
   return (
     <div className={styles.pagination}>
-      {Array.from({ length: totalPages }).map((_, index) => (
+      <button
+        className={classNames(styles.pageButton, { [styles.disabled]: currentPage === 1 })}
```

Comment:
атрибута disabled будет достаточно 
```suggestion
        className={styles.pageButton}
```

Reply:
понял)